crates/bevy_gizmos/src/config.rs

```diff
@@ -0,0 +1,163 @@
+//! A module for the [`GizmoConfig<T>`] [`Resource`].
+
+use crate as bevy_gizmos;
+pub use bevy_gizmos_macros::GizmoConfigGroup;
+
+use bevy_ecs::{component::Component, system::Resource};
+use bevy_reflect::{Reflect, TypePath};
+use bevy_render::view::RenderLayers;
+use bevy_utils::HashMap;
+use core::panic;
+use std::{
+    any::TypeId,
+    ops::{Deref, DerefMut},
+};
+
+/// A trait used to create gizmo configs groups.
+///
+/// Here you can store additional configuration for you gizmo group not covered by [`GizmoConfig`]
+///
+/// Make sure to derive [`Default`] + [`Reflect`] and register in the app using `app.init_gizmo_group::<T>()`
+pub trait GizmoConfigGroup: Reflect + TypePath + Default {}
+
+/// The default gizmo config group.
+#[derive(Default, Reflect, GizmoConfigGroup)]
+pub struct DefaultGizmoGroup;
+
+/// A [`Resource`] storing [`GizmoConfig`] and [`GizmoConfigGroup`] structs
+///
+/// Use `app.init_gizmo_group::<T>()` to register a custom config group.
+#[derive(Resource, Default)]
+pub struct GizmoConfigStore {
+    // INVARIANT: must map TypeId::of::<T>() to correct type T
+    store: HashMap<TypeId, (GizmoConfig, Box<dyn Reflect>)>,
+}
+
+impl GizmoConfigStore {
+    /// Returns [`GizmoConfig`] and [`GizmoConfigGroup`] associated with [`TypeId`] of a [`GizmoConfigGroup`]
+    pub fn get_dyn(&self, config_type_id: &TypeId) -> Option<(&GizmoConfig, &dyn Reflect)> {
+        let (config, ext) = self.store.get(config_type_id)?;
+        Some((config, ext.deref()))
+    }
+
+    /// Returns [`GizmoConfig`] and [`GizmoConfigGroup`] associated with [`GizmoConfigGroup`] `T`
+    pub fn get<T: GizmoConfigGroup>(&self) -> (&GizmoConfig, &T) {
```

Comment:
I would have this return an `Option<(&GizmoConfig, &T)>`. Can always let the end-user do the `unwrap`.

Reply:
I thought about returning ```Option<(&GizmoConfig, &T)>``` but decided to use a panicking API for the generic getter methods. Contrary to the ```get_dyn``` methods, the generic ones will only fail if the config isn't correctly registered as the type is already checked to be a gizmo config. The possibility of a plugin author failing to register the config or the user forgetting to add the plugin doesn't feel like something we need to be aware of every time the get method is used.

Even if these return an option, the ```Gizmos<T>``` SystemParam would still fail to be built when the config isn't registered.  I believe both generic access patterns should react the same way. 

But I'm open to changing it. I just wanted to share my thoughts :)

Reply:
I just don't like the panicking API on a `get` method. In the std, `get` is for non-panicking indexing. My beef is more with the name than the fact it panics. What about `config`? In fact, it doesn't really make sense to return an `Option`, panicking is the correct behavior here…

Reply:
You are right, `get` is an unfortunate name. I'm changing it to `config` for now, but maybe we will find a better name in future.